web-api/src/error/common.rs

```diff
@@ -137,10 +137,14 @@ pub(crate) struct InvalidDocumentTags {
 
 impl_application_error!(InvalidDocumentTags => BAD_REQUEST, INFO);
 
-/// Malformed document snippet: {0}
 #[derive(Debug, Error, Display, Serialize)]
-#[serde(transparent)]
-pub(crate) struct InvalidDocumentSnippet(pub(crate) InvalidString);
+#[serde(rename_all = "snake_case")]
+pub(crate) enum InvalidDocumentSnippet {
+    /// Malformed document snippet: {0}
+    InvalidString(InvalidString),
+    /// Input document didn't yield any snippets
+    NoSnippets {},
```

Comment:
❓ with the current error impl this would be a bad request on info log level. if the cutter fails, wouldn't that be more like an internal error, or at least warn log level?

Reply:
no it's not cutter failing but there not being any semantic sentences (which could be because of a subtle failure, but likely is because of different problems).

E.g. `"--------"` isn't a single valid sentence but a non-empty input. 

Note that I didn't check if `cutters` would just return that verbatim, because it doesn't matter because we do replace it very soon anyway.

Reply:
Stuff like that is why I last week asked in the channel if we could change terminology to not use splits, because at least long term we don't just split the document. We extract sentences from it. Which might happen to be splits but could very well be something else, and they definitely need all formatting stripped when we get PDFs or similar etc.